src/dll/edbgEcmdDll.C

```diff
@@ -1675,12 +1681,209 @@ uint32_t dllQueryHostMemInfoRanges( const std::vector<ecmdChipTarget> & i_target
+
+enum IplMode_t
+{
+    IPL_MODE_NORMAL = 0,
+    IPL_MODE_ISTEP = 1,
+    IPL_MODE_UNKNOWN = 0xff,
+};
+
+static edbgIPLTable g_edbgIPLTable;
+/**
+  * @brief Set the IPL mode
+  *
+  * @param enum ipl_mode i_mode - IPL mode (Interactive or Continous IPL mode)
+  *
+  * @return Upon success, ECMD_SUCCESS will be returned.  A reason code will
+  *         be returned if the execution fails.
+  */
+uint32_t setIplMode(enum IplMode_t i_mode)
+{
+    uint32_t rc = ECMD_SUCCESS;
+    
+    /* If IPL mode is set to interactive then, SBE will be set in ISTEP mode
+     * and execute all SBE isteps in chip op mode */
+    if (i_mode == IPL_MODE_ISTEP)
+    {
+        ecmdChipTarget target;
+        constexpr uint32_t MBOX_SCRATCH_REG3 = 0x283a;
+        constexpr uint32_t MBOX_SCRATCH_REG8 = 0x283f;
+        ecmdDataBuffer dataBuffer;
+
+        target.chipTypeState = ECMD_TARGET_FIELD_VALID;
+        target.cageState = target.nodeState = target.slotState = target.posState = ECMD_TARGET_FIELD_VALID;
+        target.cage = target.node = target.slot = 0;
+
+        //This will be actually set by sbe_config_update istep procedure.
+        //For testing purpose, we are setting it up.
+        dataBuffer.setWordLength(1);
+        dataBuffer.setWord(0, 0x80000000);
+        rc = dllPutCfamRegister(target, MBOX_SCRATCH_REG3, dataBuffer);
+        dataBuffer.setWord(0, 0x20000000);
+        rc = dllPutCfamRegister(target, MBOX_SCRATCH_REG8, dataBuffer);
+        if (rc)
+        {
+            return out.error(rc, FUNCNAME,
+                      "Failed dllPutCfamRegister()\n");
+        }
+    }
+    return rc;
+}
+
+/**
+  * @brief This is a helper function to call the pdbg interface
+  *        to execute an istep
+  *
+  * @param uint16_t i_start_index - position of the istep
+  * @param uint16_t i_major - Major number to execute
+  * @param uint16_t i_minor_start - Minor number start
+  * @param uint16_t i_minor_end   - Minor number end
+  *
+  * @return Upon success, ECMD_SUCCESS will be returned.  A reason code will
+  *         be returned if the execution fails.
+  */
+uint32_t iStepsHelper(uint16_t i_start_index, uint16_t i_major,
+                         uint16_t i_minor_start,
+                         uint16_t i_minor_end)
+{
+    uint32_t rc = ECMD_SUCCESS;
+    std::string IStepName;
+
+    edbgIPLTable::edbgIStepDestination_t l_destination =
+                      edbgIPLTable::EDBG_ISTEP_INVALID_DESTINATION;
+
+    for (uint16_t istep = i_minor_start; istep <= i_minor_end ; istep++) {
+        g_edbgIPLTable.getIStepNameOf(i_start_index, IStepName);
+        l_destination = g_edbgIPLTable.getDestination(i_start_index++);
+
+        //This istep is NOOP
+        if ( l_destination == edbgIPLTable::EDBG_ISTEP_NOOP ) {
+            out.print("Requested istep %s is NOOP\n", IStepName.c_str());
+        } 
+        else 
+        {
+            /* kick off start_sbe() */
+            rc = ipl_run_major_minor(i_major, istep);
```

Comment:
Need to handle Poweron step related changes here.

Reply:
I have written the poweron function but it is not a good idea to have this function in edbg. better would be to have it in libipl and call the poweron api here

Reply:
I have implemented in the istep plugin code. i will push the code